catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/NsiliEndpoint.java

```diff
@@ -106,25 +107,41 @@ public int getMaxNumResults() {
         return maxNumResults;
     }
 
+    private void setLibraryQuerySources(Set<String> querySources) {
+        Optional.ofNullable(library) .ifPresent(l -> l.setQuerySources(querySources));
+    }
+
     public void setMaxNumResults(int maxNumResults) {
         this.maxNumResults = maxNumResults;
-        if (library != null) {
-            library.setMaxNumResults(maxNumResults);
-        }
+        Optional.ofNullable(library) .ifPresent(l -> l.setMaxNumResults(maxNumResults));
     }
 
-    public List<String> getQuerySources() {
+    @Override
+    public Set<String> getQuerySources() {
         return querySources;
     }
 
-    public void setQuerySources(List<String> querySources) {
+    public void setQuerySources(Set<String> querySources) {
         this.querySources.clear();
         if (querySources != null) {
             this.querySources.addAll(querySources);
+            setLibraryQuerySources(querySources);
+        }
+    }
+
+    @Override
+    public void addQuerySource(String sourceId) {
+        if (querySources != null && !querySources.contains(sourceId)) {
```

Comment:
I think these checks are un-necessary as querySources is initialized and a set. `sourceId` could be checked for null though before attempting to add to the set.